pinot-core/src/main/java/org/apache/pinot/core/segment/creator/impl/SegmentDictionaryCreator.java

```diff
@@ -211,6 +203,39 @@ public void build()
     }
   }
 
+  private void writeBytesValueDictionary(int numValues, byte[][] sortedByteArrays) throws IOException {
+
+    if (_useVarLengthDictionary) {
+      // Backward-compatible: index file is always big-endian
+      long size = VarLengthBytesValueReaderWriter.getRequiredSize(sortedByteArrays);
+      try (PinotDataBuffer dataBuffer = PinotDataBuffer
+          .mapFile(_dictionaryFile, false, 0, size, ByteOrder.BIG_ENDIAN,
+              getClass().getSimpleName());
+          VarLengthBytesValueReaderWriter writer = new VarLengthBytesValueReaderWriter(
+              dataBuffer)) {
+        writer.init(sortedByteArrays);
```

Comment:
Any reason why the `init` does the `write` as well (comparing with existing implementation)? This caller side code becomes less readable, as readers will question why only init is called, and then go to `init` and figure out that init is also writing the values.

Reply:
I've renamed `init()` to `write()` to be clear. Please see my above comment for more details.